sentry_sdk/integrations/opentelemetry/sampler.py

```diff
@@ -0,0 +1,117 @@
+
+    # Only inherit sample rate if `traceId` is the same
+    if is_span_context_valid and parent_context.trace_id == trace_id:
+        # this is getSamplingDecision in JS
+        if parent_context.trace_flags.sampled:
+            return True
+
+        dropped = parent_context.trace_state.get(SENTRY_TRACE_STATE_DROPPED) == "true"
+        if dropped:
+            return False
+
+        # TODO-anton: fall back to sampling decision in DSC (for this die DSC needs to be set in the trace_state)
+
+    return None
+
+
+def dropped(trace_state=None):
+    # type: (Optional["TraceState"]) -> SamplingResult
+    updated_trace_context = trace_state or TraceState()
+    updated_trace_context = updated_trace_context.update(
+        SENTRY_TRACE_STATE_DROPPED, "true"
+    )
+    return SamplingResult(
+        Decision.DROP,
+        trace_state=updated_trace_context,
+    )
+
+
+class SentrySampler(Sampler):
+    def should_sample(
+        self,
+        parent_context: Optional["Context"],
+        trace_id: int,
+        name: str,
+        kind: Optional[SpanKind] = None,
+        attributes: Attributes = None,
+        links: Optional[Sequence["Link"]] = None,
+        trace_state: Optional["TraceState"] = None,
+    ) -> "SamplingResult":
+        client = sentry_sdk.get_client()
+
+        parent_span = trace.get_current_span(parent_context)
+        parent_context = parent_span and parent_span.get_span_context()
+
+        # No tracing enabled, thus no sampling
+        if not has_tracing_enabled(client.options):
+            return dropped(parent_context.trace_state)
+
+        sample_rate = None
+
+        # Check if sampled=True was passed to start_transaction
+        # TODO-anton: Do we want to keep the start_transaction(sampled=True) thing?
+
+        # Check if there is a traces_sampler
+        # Traces_sampler is responsible to check parent sampled to have full transactions.
+        has_traces_sampler = callable(client.options.get("traces_sampler"))
+        if has_traces_sampler:
+            # TODO-anton: Make proper sampling_context
+            sampling_context = {
+                "transaction_context": {
+                    "name": name,
+                },
+                "parent_sampled": parent_context.trace_flags.sampled,
+            }
+
+            sample_rate = client.options["traces_sampler"](sampling_context)
+
+        else:
+            # Check if there is a parent with a sampling decision
+            parent_sampled = get_parent_sampled(parent_context, trace_id)
+            if parent_sampled is not None:
+                sample_rate = parent_sampled
+            else:
+                # Check if there is a traces_sample_rate
+                sample_rate = client.options.get("traces_sample_rate")
+
+        # If we still have no sample rate (but enable_tracing==True,
+        # because has_tracing_enabled returned True above), default to 1.0
+        if sample_rate is None:
+            sample_rate = 1.0
```

Comment:
this is not correct, `has_tracing_enabled` is a combination of `enable_tracing`, `traces_sample_rate` and `traces_sampler` and we convert `enable_tracing` to `traces_sample_rate` [here](https://github.com/getsentry/sentry-python/blob/ea2b2931c5ec816800d21307528111275525070a/sentry_sdk/client.py#L120-L121)

so please remove these 2 lines

Reply:
done.